RadixWallet/Features/AccountRecoveryScan/Children/AccountRecoveryScanInProgress/AccountRecoveryScanInProgress.swift

```diff
@@ -0,0 +1,416 @@
+		} catch {
+			let errorMsg = "Failed to calculate next derivation paths"
+			loggerGlobal.error(.init(stringLiteral: errorMsg))
+			assertionFailure(errorMsg)
+			return .send(.delegate(.failedToDerivePublicKey))
+		}
+		let factorSourceOption: DerivePublicKeys.State.FactorSourceOption
+
+		switch state.mode {
+		case let .factorSourceWithID(_, loadableState):
+			switch loadableState {
+			case let .success(factorSource):
+				factorSourceOption = .specific(factorSource)
+			default:
+				let errorMsg = "Discrepancy! Expected to loaded the factor source"
+				loggerGlobal.error(.init(stringLiteral: errorMsg))
+				assertionFailure(errorMsg)
+				return .none
+			}
+		case let .privateHD(privateHDFactorSource):
+			factorSourceOption = .specificPrivateHDFactorSource(privateHDFactorSource)
+		}
+
+		state.status = .derivingPublicKeys
+		state.destination = .derivePublicKeys(.init(
+			derivationPathOption: .knownPaths(
+				Array(derivationPaths),
+				networkID: state.networkID
+			),
+			factorSourceOption: factorSourceOption,
+			purpose: .createEntity(kind: .account)
+		))
+
+		return .none
+	}
+
+	private func scanOnLedger(
+		accounts: IdentifiedArrayOf<Profile.Network.Account>,
+		state: inout State
+	) -> Effect<Action> {
+		assert(accounts.count == batchSize)
+		state.status = .scanningNetworkForActiveAccounts
+		state.destination = nil
+		return .run { send in
+			let (active, inactive) = try await performScan(accounts: accounts)
+			await send(.internal(.foundAccounts(active: active, inactive: inactive)))
+		}
+	}
+
+	private func performScan(
+		accounts: IdentifiedArrayOf<Profile.Network.Account>
+	) async throws -> (
+		active: IdentifiedArrayOf<Profile.Network.Account>,
+		inactive: IdentifiedArrayOf<Profile.Network.Account>
+	) {
+		let accountAddresses: [AccountAddress] = accounts.map(\.address)
+
+		do {
+			let activeAccounts = try await onLedgerEntitiesClient
+				.getAccounts(
+					accountAddresses,
+					// actually we wanna `resourceMetadataKeys` as well here, but we cannot since
+					// the count will exceed `EntityMetadataKey.maxAllowedKeys`.
+					metadataKeys: [.ownerBadge, .ownerKeys],
+					forceRefresh: true
+				)
+				.filter { (onLedgerAccount: OnLedgerEntity.Account) -> Bool in
+
+					guard
+						case let metadata = onLedgerAccount.metadata,
+						let ownerKeys = metadata.ownerKeys,
+						let ownerBadge = metadata.ownerBadge
+					else { return false }
+
+					func hasStateChange(_ list: OnLedgerEntity.Metadata.ValueAtStateVersion<some Any>) -> Bool {
+						list.lastUpdatedAtStateVersion > 0
+					}
+					let isActive = hasStateChange(ownerKeys) || hasStateChange(ownerBadge)
+					return isActive
+				}
+
+			var active: IdentifiedArrayOf<Profile.Network.Account> = []
+			var inactive: IdentifiedArrayOf<Profile.Network.Account> = []
+			for account in accounts {
+				if let activeAccount = activeAccounts.first(where: { $0.address == account.address }) {
+					var account = account
```

Comment:
use `if var` instead?

Reply:
`activceAccount` is never modified, `account` is. and `for var account in accounts` is slightly less precise, since it looks like we are changing account when it is active too, so what I have here is most precise.